contracts/extensions/PersonalEditionExtension.sol

```diff
@@ -0,0 +1,125 @@
+pragma solidity 0.4.21;
+import "../Extension.sol";
+import "../GnosisSafe.sol";
+
+
+/// @title Personal Edition Extension
+/// @author Richard Meissner - <richard@gnosis.pm>
+contract PersonalEditionExtension is Extension {
+
+    string public constant NAME = "Personal Edition Extension";
+    string public constant VERSION = "0.0.1";
+
+    PersonalEditionExtension masterCopy;
+    GnosisSafe gnosisSafe;
+
+    uint256 public nonce;
+
+    modifier onlyGnosisSafe() {
+        require(msg.sender == address(gnosisSafe));
+        _;
+    }
+
+    /// @dev Allows to upgrade the contract. This can only be done via a Safe transaction.
+    /// @param _masterCopy New contract address.
+    function changeMasterCopy(PersonalEditionExtension _masterCopy)
+        public
+        onlyGnosisSafe
+    {
+        require(address(_masterCopy) != 0);
+        masterCopy = _masterCopy;
+    }
+
+    /// @dev Function to be implemented by extension. This is used to check to what Safe the Extension is attached.
+    /// @return Returns the safe the Extension is attached to.
+    function getGnosisSafe()
+        public
+        returns (GnosisSafe)
+    {
+        return gnosisSafe;
+    }
+
+    /// @dev Setup function sets initial storage of contract.
+    function setup()
+        public
+    {
+        // gnosisSafe can only be 0 at initalization of contract.
+        // Check ensures that setup function can only be called once.
+        require(address(gnosisSafe) == 0);
+        gnosisSafe = GnosisSafe(msg.sender);
+    }
+
+    /// @dev Allows to execute a Safe transaction confirmed by required number of owners.
+    /// @param to Destination address of Safe transaction.
+    /// @param value Ether value of Safe transaction.
+    /// @param data Data payload of Safe transaction.
+    /// @param operation GnosisSafe.Operation type of Safe transaction.
+    /// @param v Array of signature V values sorted by owner addresses.
+    /// @param r Array of signature R values sorted by owner addresses.
+    /// @param s Array of signature S values sorted by owner addresses.
+    function payAndExecuteTransaction(address to, uint256 value, bytes data, GnosisSafe.Operation operation, uint8[] v, bytes32[] r, bytes32[] s, uint256 price, uint8 pv, bytes32 pr, bytes32 ps)
+        public
+    {
+        bytes32 priceHash = getPriceHash(msg.sender, price, nonce);
```

Comment:
An alternative would be to have this as part of the normal hash (getTransactionHash) which would reduce the costs, as we would have less parameters and less checks.